flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/plan/rules/logical/PushProjectIntoTableSourceScanRuleTest.java

```diff
@@ -107,4 +96,13 @@ private void testNestedProject(boolean nestedProjectionSupported) {
 		util().verifyPlan(sqlQuery);
 	}
 
+	@Test
+	public void testComplicatedNestedProject() {
+		String sqlQuery = "SELECT id," +
+				"    deepNested.nested1.name AS nestedName,\n" +
+				"    deepNested.nested2 AS nested2,\n" +
+				"    deepNested.nested2.num AS nestedNum\n" +
+				"FROM NestedTable";
+		util().verifyPlan(sqlQuery);
```

Comment:
nit: add a test about complex expressions, such as  deepNested.nested1.name + nested.value

Reply:
Add test with calculation in query.